spec/puppet-server-lib/puppet/jvm/ast_compiler_spec.rb

```diff
@@ -135,5 +149,41 @@ def find_notify(catalog, title: nil)
         end
       end
     end
+
+    context 'when running a project plan' do
+      let(:boltlib_path) { [] }
+      context 'single threaded' do
```

Comment:
I notice this PR no longer has any special handling for multithreaded. If we're not going to test that mode, but we still sort of expect this to work there, I think we should drop this context wrapper. Fwiw I don't think there's a super good way to actually test it in that mode, so I'm fine not having it, but having this context makes it look like there's something special about single threaded.

If, however, we _don't_ think this works in multithreaded, we should perhaps be more aggressive about preventing people from using it in that mode.

Reply:
Deleted the context and the redundant settings expectations. This is currently blocking a bunch of other work. Can we ticket out investigating the multithreaded bit so we can get this merged and start really testing it?

Reply:
Oh yeah, I don't think it's very important right this minute, I just wanted to understand what the current expectation is.